src/core/Akka.Cluster/Member.cs

```diff
@@ -468,16 +468,21 @@ public override int GetHashCode()
         /// <summary>
         /// TBD
         /// </summary>
-        /// <param name="that">TBD</param>
+        /// <param name="uniqueAddress">TBD</param>
         /// <returns>TBD</returns>
-        public int CompareTo(UniqueAddress that)
+        public int CompareTo(UniqueAddress uniqueAddress)
+        {
+            if (uniqueAddress == null) throw new ArgumentNullException(nameof(uniqueAddress));
+
+            var result = Address.Comparer.Compare(Address, uniqueAddress.Address);
```

Comment:
This changes how `UniqueAddress`es are compared - previously a protocol and system name parts were ignored. I think it was a bug, and it's not like the JVM version was working. It should be correct now.